src/marqo/core/search/recommender.py

```
@@ -0,0 +1,165 @@
+                valid_tensor_fields = marqo_index.tensor_field_map.keys()
+                for tensor_field in tensor_fields:
+                    if tensor_field not in valid_tensor_fields:
+                        raise InvalidFieldNameError(f'Tensor field "{tensor_field}" not found in index "{index_name}". '
+                                         f'Available tensor fields: {", ".join(valid_tensor_fields)}')
+
+        if isinstance(documents, dict):
+            document_ids = list(documents.keys())
+        else:
+            document_ids = documents
+
+        t0 = timer()
+
+        marqo_documents = tensor_search.get_documents_by_ids(
+            config.Config(self.vespa_client),
+            index_name, document_ids, show_vectors=True
+        )
+
+        # Make sure all documents were found
+        not_found = []
+        for document in marqo_documents['results']:
+            if not document['_found']:
+                not_found.append(document['_id'])
+
+        if len(not_found) > 0:
+            raise InvalidArgumentError(f'The following document IDs were not found: {", ".join(not_found)}')
+
+        doc_vectors: Dict[str, List[List[float]]] = {}
+        for document in marqo_documents['results']:
+            vectors: List[List[float]] = []
+            for tensor_facet in document['_tensor_facets']:
+                field = list(tensor_facet.keys())[0]
+                if tensor_fields is None or field in tensor_fields:
+                    vectors.append(tensor_facet['_embedding'])
+
+            doc_vectors[document['_id']] = vectors
+
+        vectors: List[List[float]] = []
+        weights: List[float] = []
+
+        for document_id, vector_list in doc_vectors.items():
+            if isinstance(documents, dict):
+                weight = documents[document_id]
+            else:
+                weight = 1
+            vectors.extend(vector_list)
+            weights.extend([weight] * len(vector_list))
+
+        if len(vectors) == 0:
```

I think this happens when provided documents do not have tensor fields? Should we raise an error here?